src/common/LogEntry.cc

```diff
@@ -313,31 +313,37 @@ void LogSummary::encode(bufferlist& bl, uint64_t features) const
     ENCODE_START(2, 2, bl);
     encode(version, bl);
     list<LogEntry> tail;
-    build_ordered_tail(&tail);
+    build_ordered_tail_legacy(&tail);
     encode(tail, bl, features);
     ENCODE_FINISH(bl);
     return;
   }
-  ENCODE_START(3, 3, bl);
+  ENCODE_START(4, 3, bl);
   encode(version, bl);
   encode(seq, bl);
   encode(tail_by_channel, bl, features);
+  encode(channel_info, bl);
+  recent_keys.encode(bl);
   ENCODE_FINISH(bl);
 }
 
 void LogSummary::decode(bufferlist::const_iterator& bl)
 {
-  DECODE_START_LEGACY_COMPAT_LEN(3, 2, 2, bl);
+  DECODE_START_LEGACY_COMPAT_LEN(4, 2, 2, bl);
   decode(version, bl);
   if (struct_v < 3) {
     list<LogEntry> tail;
     decode(tail, bl);
     for (auto& i : tail) {
-      add(i);
+      add_legacy(i);
```

Comment:
i think we can drop the v3 dencoder for `LogSummary`. see #42276. will update my cleanup once this PR is merged.